src/core/cgroup.c

```diff
@@ -817,9 +818,9 @@ int cgroup_log_xattr_apply(Unit *u, const char *cgroup_path) {
         if (!patterns)
                 return log_oom_debug();
 
-        memcpy_safe(patterns, allowed_patterns, allowed_patterns_len);
-        patterns[allowed_patterns_len] = '\xff';
-        memcpy_safe(&patterns[allowed_patterns_len + 1], denied_patterns, denied_patterns_len);
+        last = mempcpy_safe(patterns, allowed_patterns, allowed_patterns_len);
+        *(last++) = '\xff';
```

Comment:
Won't this increment `last` first and then dereference it? We need to dereference last first and then increment it to have correct behavior.

Reply:
It's the postfix increment, so the actual dereferenced value will be the previous value of `last`.